src/Compilers/Core/Portable/Operations/OperationKind.cs

```diff
@@ -149,7 +149,9 @@ public enum OperationKind
         DynamicIndexerAccessExpression = 0x128,
         /// <summary>Indicates an <see cref="ITranslatedQueryExpression"/>.</summary>
         TranslatedQueryExpression = 0x129,
-
+        /// <summary>Indicates an <see cref="IDeconstructionAssignmentExpression"/>.</summary>
+        DeconstructionAssignmentExpression = 0x12a,
+        
         // Expressions that occur only in C#.
```

Comment:
>// Expressions that occur only in C#. [](start = 8, length = 37)

What is the purpose of maintaining ranges per language? What if other language starts supporting the same expression? Should DeconstructionAssignmentExpression be placed into C#-only region?

Reply:
> What is the purpose of maintaining ranges per language? 

This is something that John had been using. We can consider unifying all such ranges.

> Should DeconstructionAssignmentExpression be placed into C#-only region?

I can move it there, though your previous question was my assumption that VB will support this soon.